### PR TITLE
Feature/chat

### PR DESCRIPTION
**Pull request by mhdbaal**
# RocketChat Iframe Integration

## Description

This is a basic RocketChat integration for Latelier, providing access to your RocketChat channels via an embedded RocketChat iframe.

In order to enable the feature, a chat settings entry must added in your settings.json file (see **public.chat** in settings-development.sample.json). 
You can also provide a default channel in the settings.

Once setup,  a new chat icon appears on the topbar right corner. By clicking on it you can toggle the display of the chat, and switch between different chat channels.

You can only add new chat channels to Latelier if you are admin (check out Administration Page). You can also bind a channel with 1 project.
A chat channel item in Latelier will only store the url of a particular RocketChat channel.

## Todo
i still need to add tests, update the README configuration and also some basic features (especially checking if a user can read a project before giving him access to the related channel)

### CODE REVIEW

File: settings-development.sample.json [hunk 1]
```diff
@@ -39,6 +39,17 @@
       "enabled": true,
       "email": "ct-remote-email"
     },
-    "uploadTransport": "ddp"
+    "uploadTransport": "ddp",
+    "chat": {
```
Comment (jbl2024) [inline settings-development.sample.json:43]: README.md should be updated with theses new settings

File: imports/ui/administration/chat/ChatChannelDetail.vue [hunk 2]
```diff
@@ -0,0 +1,179 @@
+<template>
+  <v-card class="chat-channel-detail">
+    <confirm-dialog
+      v-if="!isNew"
+      :active.sync="showConfirmDeleteChatChannelDialog"
+      title="Confirmer la suppression ?"
```
[the remark this answers is not in the document]
Comment (mhdbaal) [reply imports/ui/administration/chat/ChatChannelDetail.vue]: Fixed in 55a9c1a269a7e74d5b84cd5220501dabb1d9f73d

File: imports/ui/administration/chat/ChatChannelDetail.vue [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  </v-card>
+</template>
+
+<script>
+import ChatProjectAutocomplete from './ChatProjectAutocomplete';
+export default {
+  components: {
+    ChatProjectAutocomplete
+  },
+  i18n: {
```
[the remark this answers is not in the document]
Comment (mhdbaal) [reply imports/ui/administration/chat/ChatChannelDetail.vue]: Fixed in 55a9c1a269a7e74d5b84cd5220501dabb1d9f73d

File: imports/ui/administration/chat/ChatChannelDetail.vue [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        this.close();
+        this.$emit("created");
+      });
+    },
+    update() {
+      const updatedChatChannel = {
+        id: this.chatChannel._id,
+        name: this.name,
+        channel: this.channel,
+        projectId: this.project && this.project._id ?this.project._id : null
```
[the remark this answers is not in the document]
Comment (mhdbaal) [reply imports/ui/administration/chat/ChatChannelDetail.vue]: Fixed in f5f303a75392b0a3702ae5c135722dd546d63cb1

File: settings-development.sample.json (hunk 1 above)
Comment (mhdbaal) [reply settings-development.sample.json:43]: Fixed in f5f303a75392b0a3702ae5c135722dd546d63cb1

**Pull request closed without merging**